test/test-pull.sh

```diff
@@ -145,6 +145,39 @@ begin_test "pull"
 )
 end_test
 
+begin_test "pull (literal --include)"
```

Comment:
This test looks correct to me based on the description of what's going wrong with the filepath patterns, but it's passing for me on master, so I don't think it's actually testing the root issue